### PR TITLE
Automated Documentation

### PR DESCRIPTION
**Pull request by kratsg**
We now have Sphinx integration to automatically generate documentation for us using:
- `Doxygen` to parse the C++ header/source files (see `docs/Doxyfile` for details)
- `Breathe` to convert the generated XML files (`doxygen/xml`) into something that `Sphinx` can use
- `ReadTheDocs` service integration to automatically build into a webpage that can be viewed here: https://xaodanahelpers.readthedocs.org/

This also supports latex rendering correctly, provided you comment it right within the doxygen + sphinx documentation..

/cc @kkrizka 

will be happy to see this support at least.

### CODE REVIEW

**Comment by kratsg** [comment]
For details of how I did it, here's an overview:

1. create a `docs` folder which all stuff will be located in
1. generate a Doxyfile with `GENERATE_XML = ON` at a minimum and point it towards an output relative (inside the current `docs` folder). Make sure you use relative paths to point towards the header and source files for doxygen to parse through
1. run `sphinx-quickstart` to generate a default `conf.py`
1. follow the [Breathe Quickstart](https://breathe.readthedocs.org/en/latest/quickstart.html) to update the `conf.py` making sure it can point the project towards doxygen's generated XML output

and that should be it. It should all work.

**Comment by kratsg** [comment]
tl;dr -- use doxygen styles to document code correctly and Sphinx styles to get the formatting right!

**Comment by kratsg** [comment]
This closes #323
